yt/frontends/boxlib/data_structures.py

```diff
@@ -29,22 +29,47 @@
 _scinot_finder = re.compile(r"[-+]?[0-9]*\.?[0-9]+([eEdD][-+]?[0-9]+)?")
 # This is the dimensions in the Cell_H file for each level
 # It is different for different dimensionalities, so we make a list
+_1dregx = r"[-]?\d+"
+_2dregx = r"[-]?\d+,[-]?\d+"
+_3dregx = r"[-]?\d+,[-]?\d+,[-]?\d+"
 _dim_finder = [
-    re.compile(r"\(\((\d+)\) \((\d+)\) \(\d+\)\)$"),
-    re.compile(r"\(\((\d+,\d+)\) \((\d+,\d+)\) \(\d+,\d+\)\)$"),
-    re.compile(r"\(\((\d+,\d+,\d+)\) \((\d+,\d+,\d+)\) \(\d+,\d+,\d+\)\)$"),
+    re.compile(r"\(\((" + _1dregx + ")\) \((" + _1dregx + ")\) \(" + _1dregx + "\)\)$"),
+    re.compile(r"\(\((" + _2dregx + ")\) \((" + _2dregx + ")\) \(" + _2dregx + "\)\)$"),
+    re.compile(r"\(\((" + _3dregx + ")\) \((" + _3dregx + ")\) \(" + _3dregx + "\)\)$"),
 ]
 # This is the line that prefixes each set of data for a FAB in the FAB file
 # It is different for different dimensionalities, so we make a list
 _endian_regex = r"^FAB \(\(\d+, \([0-9 ]+\)\),\((\d+), \(([0-9 ]+)\)\)\)"
 _header_pattern = [
-    re.compile(_endian_regex + r"\(\((\d+)\) \((\d+)\) \((\d+)\)\) (\d+)\n"),
```

Comment:
This whole block could be reformatted as follow:
```suggestion
_endian_regex = r"^FAB\ \(\(\d+,\ \([0-9\ ]+\)\),\((\d+),\ \(([0-9\ ]+)\)\)\)"
_header_pattern = [
    re.compile(
        rf"""{_endian_regex}           # match `endianness` (??)
        \(
              \(( {ndregx} )\)         # match `start` (??)
            \ \(( {ndregx} )\)         # match `end` (??)
            \ \(( {ndregx} )\)         # match `centering` (??)
        \) 
        \ (-?\d+)                      # match `nc` (??)
        $ # end of line
        """,
        re.VERBOSE
    )
    for ndregx in (_1dregx, _2dregx, _3dregx)    
]
```
If you understand what each of these blobs match, it would be awesome if you could expand a bit the comments (where I left `??`).

Reply:
Oh! thank you for explaining this! No wonder when I tried `re.VERBOSE` locally it did not work for me. I did not know that spaces need an escape \ before them when using `VERBOSE`

Reply:
Thank you for this suggestion. I am not able to directly commit this through the github interface as it says "applying suggestions on deleted lines in not supported" So I am going to manually add this to the branch.